### PR TITLE
Properly register text_direction_codepoint_in_comment lint.

### LINKED ISSUE #90614: It's not possible to allow #![allow(text_direction_codepoint_in_comment)]
I tried this code:

```rust
#![allow(text_direction_codepoint_in_comment)]
```

I expected to see this happen: `text_direction_codepoint_in_comment` to be allowed

Instead, this happened:

```
warning: unknown lint: `text_direction_codepoint_in_comment`
 --> src/main.rs:1:10
  |
1 | #![allow(text_direction_codepoint_in_comment)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: did you mean: `text_direction_codepoint_in_literal`
  |
  = note: `#[warn(unknown_lints)]` on by default
```

(in case anyone is curious why I noticed this, don't worry, this isn't due to production code, see https://codegolf.stackexchange.com/a/230964/3103)

### Meta

`rustc --version --verbose`:
```
rustc 1.56.1 (59eed8a2a 2021-11-01)
binary: rustc
commit-hash: 59eed8a2aac0230a8b53e89d4e99d55912ba6b35
commit-date: 2021-11-01
host: x86_64-unknown-linux-gnu
release: 1.56.1
LLVM version: 13.0.0
```

### PR DESCRIPTION
**Pull request by hkratz**
This makes it known to the compiler so it can be configured like with `#![allow(text_direction_codepoint_in_comment)]`.

Fixes #90614.

### CODE REVIEW

File: src/test/ui/lint/issue-90614-accept-allow-text-direction-codepoint-in-comment-lint.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// build-pass
```
Comment (jyn514) [inline src/test/ui/lint/issue-90614-accept-allow-text-direction-codepoint-in-comment-lint.rs:1]:
```suggestion
// check-pass
```

**Review by jyn514 — CHANGES_REQUESTED** [review]

File: src/test/ui/lint/issue-90614-accept-allow-text-direction-codepoint-in-comment-lint.rs (hunk 1 above)
Comment (hkratz) [reply src/test/ui/lint/issue-90614-accept-allow-text-direction-codepoint-in-comment-lint.rs]: Squashed in.

**Comment by jyn514** [comment]
Thanks!

@bors r+ rollup

**Comment by apiraino** [comment]
Beta backport approved as per compiler team [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202021-11-11.20.2354818/near/261145463)

@rustbot label +beta-accepted

**Pull request merged**